pkg/build/util/util.go

```diff
@@ -238,17 +239,19 @@ func SafeForLoggingURL(u *url.URL) *url.URL {
 	if u == nil {
 		return nil
 	}
-	newUrl := *u
-	newUrl.User = url.User("redacted")
-	return &newUrl
+	newURL := *u
```

Comment:
Even though you added a deepcopy higher in the stack, there's no guarantee every caller to SafeForLoggingURL is coming in via that path (in fact I don't think they all are, see: SafeForLoggingS2IConfig for example)

So I think it's still important to deepcopy the url arg here, we shouldn't be mutating the input to the function.